sycl-fusion/passes/kernel-fusion/SYCLKernelFusion.cpp

```diff
@@ -170,27 +168,25 @@ void SYCLKernelFusion::fuseKernel(
   // Fuction names may appear multiple times, resulting in
   // the corresponding function to be included multiple times. The functions
   // are included in the same order as given by the list.
-  llvm::MDNode *MD = StubFunction.getMetadata(MetadataKind);
+  MDNode *MD = StubFunction.getMetadata(MetadataKind);
   assert(MD && MD->getNumOperands() == 2 &&
          "Metadata tuple should have two operands");
 
-  // Name of the fused kernel.
-  auto *KernelName = dyn_cast<MDString>(MD->getOperand(0).get());
+  // First metadata operand should be the fused kernel name.
+  auto *KernelName = cast<MDString>(MD->getOperand(0).get());
   assert(KernelName &&
```

Comment:
Just a note that this assert isn't needed anymore, because it exists within `cast`: that one will fail first and we won't see our custom message anyway.